### ISSUE TITLE
Foreign Key Remapping Causes Confusion

### ISSUE THREAD

**Issue by MarkRx** [metabase/metabase#8599]
TLDR - Foreign key remapping isn't being applied to lists of values - see https://github.com/metabase/metabase/issues/8599#issuecomment-425918367

The ability to remap foreign keys to values was added in Metabase 0.30.x. It's a pretty nice feature. However, after having seen people use them I have noticed a few things that trip people up:

1. Filters on foreign key fields still operate on the FK ID instead of the mapped value. For example in the sample dataset if the "Orders" "User ID" is mapped to a "User" "Name" the name will show up as the value but filters will operate on "User ID". Since people see an entity value when a field is mapped they tend to think they can just filter on that field directly based on the value they see.
2. Similar to 1 - Naming FK column names is difficult. After setting up FK remappings admins have a choice to keep the name of the column (say User ID) or rename it to match an entity (say User). If they keep the name User ID then the result set is confusing because it shows a "Name". If they choose to name it "User" then the filter is confusing because it operates on an ID.

![image](https://user-images.githubusercontent.com/17504314/46170366-98769e80-c26b-11e8-9db1-7f3a942348f8.png)

The above image demonstrates how a user would inuitively try to filter on "Gerald" when really they should filter on the ID of "123". This is because the field was called "User" and is mapped to the username in the user table thus showing up as a name in query results.

![user_filter](https://user-images.githubusercontent.com/17504314/46172179-cf9b7e80-c270-11e8-9870-9204e0563e05.PNG)

The above image image demonstrates how a user could be confused when deciding how to filter on "User"

What is the recommended way to address this problem? Use a naming convention FKs to make things more apparent (~~like ID~~ Metabase seems to strip off the text "ID" at the end of a field name)? Or should filtering be updated to automatically filter on FK mappings? If the later would people ever want to really filter on ID and not on some mapped value?

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by MarkRx** [comment]
I can't find the code but something is stripping off "ID" if a field ends in ID. This results in confusion because a filter on "User" is not a filter on a User entity... it's a filter on a FK (typically a number).

**Comment by flamber** [comment]
Is this related to or a duplicate of #6179 ?

**Comment by MarkRx** [comment]
#6179 is related. This issue morphed into discovering that FK remapping isn't being done for lists of values - see https://github.com/metabase/metabase/issues/8599#issuecomment-425918367.

**Comment by 2Ben** [comment]
I have what I think is a related issue: FK mapping seems to only work partially, for some reason:
![image](https://user-images.githubusercontent.com/876405/53691125-92549300-3db2-11e9-9045-c6e41d5eddf5.png)

The schema is like this:

Service Request -FK-> SysAid User -FK-> Company

I want to filter SR based on Company.

**Comment by ranquild** [comment]
https://github.com/metabase/metabase/issues/8599#issuecomment-425918367
this is fixed and tested extensively in https://github.com/metabase/metabase/pull/45047
